File: model/src/main/java/io/mateu/erp/model/world/State.java
package io.mateu.erp.model.world;

import io.mateu.erp.model.product.transfer.TransferPoint;
import io.mateu.ui.mdd.server.annotations.Ignored;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;

/**
 * holder for states
 *
 * Created by miguel on 13/9/16.
 */
@Entity
@Getter@Setter
public class State {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne
    @NotNull
    private Country country;

    @NotNull
    private String name;

    @OneToMany(mappedBy = "state")
    @Ignored
    private List<City> cities = new ArrayList<>();

    @OneToMany(mappedBy = "gatewayOf")
    @Ignored
    private List<TransferPoint> gateways = new ArrayList<>();

}

File: model/src/main/java/io/mateu/erp/model/world/State.kt
package io.mateu.erp.model.world

import io.mateu.erp.model.product.transfer.TransferPoint
import io.mateu.ui.mdd.server.annotations.Ignored

import javax.persistence.*
import javax.validation.constraints.NotNull
import java.util.ArrayList

/**
 * holder for states
 *
 * Created by miguel on 13/9/16.
 */
@Entity
class State {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @ManyToOne
    @NotNull
    var country: Country? = null

    @NotNull
    var name: String? = null

    @OneToMany(mappedBy = "state")
    @Ignored
    var cities = ArrayList<City>()

    @OneToMany(mappedBy = "gatewayOf")
    @Ignored
    var gateways = ArrayList<TransferPoint>()

}
